x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java

```diff
@@ -94,10 +145,20 @@ private void doAutoFollow() {
             return;
         }
 
-        Consumer<Exception> handler = e -> {
-            if (e != null) {
-                LOGGER.warn("failure occurred during auto-follower coordination", e);
+        Consumer<List<AutoFollowResult>> handler = results -> {
+            for (AutoFollowResult result : results) {
+                if (result.clusterStateFetchException != null) {
```

Comment:
This flow is quite similar to `updateStats`. Should we combine these to a single method?